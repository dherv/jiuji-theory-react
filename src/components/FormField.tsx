import { Field } from 'formik';
import styled from 'styled-components';
import { themeBlueField } from '../styled/themes';

const FormField = styled(Field)`
  padding: 0.5rem;
  font-weight: 500;
  font-size: 12px;
  border: ${(props) => {
    if (props.error && props.touched) return '1px solid #ff867c';
    if (props.touched) return '1px solid #98ee99';
    return props.theme.border;
  }};
`;

FormField.defaultProps = {
  theme: themeBlueField,
};

export default FormField;
