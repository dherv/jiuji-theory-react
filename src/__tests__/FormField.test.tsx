import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import FormField from "../components/FormField";
import { Formik, Form } from "formik";

const props = {
  onChange: jest.fn(),
  placeholder: "test",
};

test("should display an FormField with a placeholder", () => {
  render(
    <Formik initialValues={{ name: "" }} onSubmit={jest.fn()}>
      <Form>
        <FormField {...props} />
      </Form>
    </Formik>
  );
  expect(screen.getByPlaceholderText("test")).toBeInTheDocument();
});

test("should call onChange when user type in the FormField", () => {
  render(
    <Formik initialValues={{ name: "" }} onSubmit={jest.fn()}>
      <Form>
        <FormField {...props} />
      </Form>
    </Formik>
  );
  fireEvent.change(screen.getByPlaceholderText("test"), {
    target: { value: "23" },
  });
  expect(props.onChange).toHaveBeenCalled();
});

test("should render the FormField correctly", () => {
  const { asFragment } = render(
    <Formik initialValues={{ name: "" }} onSubmit={jest.fn()}>
      <Form>
        <FormField {...props} />
      </Form>
    </Formik>
  );
  expect(asFragment()).toMatchSnapshot();
});
