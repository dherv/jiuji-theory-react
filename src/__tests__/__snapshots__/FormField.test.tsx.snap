// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the FormField correctly 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <input
      class="sc-AxjAm jMUqqh"
      placeholder="test"
      theme="[object Object]"
      value="[object Object]"
    />
  </form>
</DocumentFragment>
`;
